src/engine/Service.cpp

```diff
@@ -592,10 +599,21 @@ void Service::precomputeSiblingResult(std::shared_ptr<Operation> left,
       // Stop precomputation as the size of `siblingResult` exceeds the
       // threshold it is not useful for the service operation. Pass the
       // partially materialized result to the sibling.
+      auto identityFunction = [](auto& p) { return std::move(p); };
+      std::vector<std::unique_ptr<Result::LazyResult>> viewCollection;
+      viewCollection.emplace_back(std::make_unique<Result::LazyResult>(
+        ad_utility::CachingTransformInputRange(
+            Result::LazyResult{std::move(resultPairs)},
```

Comment:
Why do you need the additional ` LazyResult` constructor here,
same question two lines below?